Rewrote ZHA docs info on device support and quirks. (#27170)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

### Knowing which devices are supported

> Home Assistant's ZHA integration supports all standard Zigbee device types and should be compatible with most Zigbee devices as long as they fully conform to the official ZCL (Zigbee Cluster Library) specifications as set by the [CSA (Connectivity Standards Alliance, formerly the Zigbee Alliance)](https://csa-iot.org/all-solutions/zigbee/). There is therefore no official compatibility list of devices that will work out-of-the-box with the ZHA integration.

Reviewer: ```suggestion
Home Assistant's ZHA integration supports all standard Zigbee device types. It should be compatible with most Zigbee devices as long as they fully conform to the official ZCL (Zigbee Cluster Library) specifications defined by the [CSA (Connectivity Standards Alliance, formerly the Zigbee Alliance)](https://csa-iot.org/all-solutions/zigbee/). There is therefore no official compatibility list of devices that will work out-of-the-box with the ZHA integration.
```

> Be aware that there are hardware manufacturers that do not always fully comply with the standard specifications or implement unique features, and as such some Zigbee devices will pair/join fine with ZHA but then only show none or only a few entities in the integration. Know that developers can work around most such interoperability issues by adding conversion/translation code in custom device handlers. Please see the section below about "How to add support for new and unsupported devices" for more information.

Reviewer: ```suggestion
Not all hardware manufacturers always fully comply with the standard specifications. Sometimes, they may also implement unique features. For this reason, some Zigbee devices pair/join fine with ZHA but then only show none or only a few entities in the integration. Developers can work around most such interoperability issues by adding conversion/translation code in custom device handlers. For more information, refer to the section below on _How to add support for new and unsupported devices_.
```

> For clarification, noramlly there will only be devices that do not fully conform to CSA's ZCL specifications that will not present all standard attributes as entities for configuration in the ZHA integration. Zigbee devices that only use the standard clusters and attributes that are Zigbee specifications set by the Connectivity Standards Alliance should not need any custom device handlers.

Reviewer: ```suggestion
For clarification, normally only devices that do not fully conform to CSA's ZCL specifications that will not present all standard attributes as entities for configuration in the ZHA integration. Zigbee devices that only use the standard clusters and attributes that are Zigbee specifications set by the Connectivity Standards Alliance should not need custom device handlers.
```

> Note! Before continuing with that section, please that if a device does not join/pair at all then your issues probably have other problems, so first be sure that you have read and followed the actions suggested in the troubleshooting sections about how to avoid pairing/connection difficulties + interference avoidance and network range/coverage optimization.

Reviewer: ```suggestion
Before continuing with this section: If a device does not join/pair at all, read the troubleshooting sections about how to avoid pairing/connection difficulties, interference avoidance, and network range/coverage optimization.
```

> Also, a tip to new Zigbee users is to look at [blakadder's unofficial Zigbee Device Compatibility Repository](https://zigbee.blakadder.com) which anyone from any home automation community can help maintain by submitting device compatibility to it. His unofficial Zigbee device database contains independent community member's reports or device-specific pairing tips and tricks for several home automation gateway/bridge/hub software, including but not limited to open-source Zigbee implementations, such as example; ZHA, Zigbee2MQTT, and Tasmota (Zigbee2Tasmota).

Reviewer: ```suggestion
Tip to new Zigbee users: Checkout [blakadder's unofficial Zigbee Device Compatibility Repository](https://zigbee.blakadder.com). Anyone can help maintain the site by submitting device compatibility information to it. The repository contains independent community member's reports or device-specific pairing tips for several home automation gateway/bridge/hub software, including open-source Zigbee implementations, such as ZHA, Zigbee2MQTT, and Tasmota (Zigbee2Tasmota).
```

#### How to add support for new and unsupported devices

> If you have already followed the recommended troubleshooting steps and your Zigbee device pairs/joins successfully with the ZHA integration but still does not show any entities or not all of the expected entities, even after trying to re-pair/re-join it several times, you may need a custom device handler. This handler will have exception code to work around specific device deviations or unique features and functions.

Reviewer: ```suggestion
If your Zigbee device pairs/joins successfully with the ZHA integration but does not show all of the expected entities: 
1. Try to re-pair/re-join the device several times.
2. Checkout the troubleshooting section.
3. Still not working? You may need a custom device handler. This handler will have exception handling code to work around device-specific issues.
```

> For devices that do not follow the standard clusters and attributes provided in the CSA's ZCL (Zigbee Cluster Library) specifications, the ZHA integration relies on a project called "[ZHA Device Handlers (also known as "zha-quirk")](https://github.com/zigpy/zha-device-handlers)". It contains device-specific Python scripts, referred to in ZHA as "quirks", that can resolve compliance and interoperability issues by implementing on-the-fly conversion/translation of custom Zigbee configurations and parameters or implementing manufacturer-specific features and functions for specific devices.

Reviewer: ```suggestion
For devices that do not follow the standard defined in the CSA's ZCL (Zigbee Cluster Library), the ZHA integration relies on a project called "[ZHA Device Handlers (also known as "zha-quirk")](https://github.com/zigpy/zha-device-handlers)". It contains device-specific Python scripts called "quirks". These scripts can resolve compliance and interoperability issues by implementing on-the-fly conversion of custom Zigbee configurations or by implementing manufacturer-specific features for specific devices.
```

People familiar with other Zigbee gateway solutions for home automation may know similar concepts of using custom Zigbee device handlers/converters for non-standard devices. For example, [Zigbee2MQTT (and IoBroker) uses zigbee-herdsman converters](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html) and [SmartThings Classics (Legacy) platform has Hub Connected Device Handlers](https://developer.smartthings.com/docs/devices/hub-connected/legacy).

> If you are not able to develop such a custom "quirk" Python script yourself with the custom handler/converter/parser/translator code needed for your specific device and add it to the "ZHA Device Handlers" repository/library, you can submit a "device support request" as a new issue to the [ZHA Device Handlers project repository on GitHub](https://github.com/zigpy/zha-device-handlers/issues). To do so, sign up/in on GitHub and click "New issue" and follow the instructions. New device support requests requires at minimum that the device owner supply the device signature + diagnostic information. The device owner may also need to actively assist in further testing or provide additional information that volunteering developers might to create custom ZHA Device Handlers for your devices.

Reviewer: ```suggestion
If you do not want to develop such a "quirk" Python script yourself, you can submit a "device support request" as a new issue to the [ZHA Device Handlers project repository on GitHub](https://github.com/zigpy/zha-device-handlers/issues): 
1. Sign in to GitHub.
2. Select **New issue** and follow the instructions. 
  * New device support requests require the device signature + diagnostic information. 
  * You may also need to actively help in further testing or provide additional information to the volunteering developers. 
```

> Note that submitting a new "device support request" does not guarantee that someone else will develop a custom "quirk" for ZHA, as it relies on volunteering developers contributing new ZHA Device Handlers to the project. However, without "device support requests" being submitted for deviating devices to the ZHA Device Handlers project, it is likely that the ZHA integration developers will not know that the specific Zigbee device is not working correctly in ZHA unless they happen to have bought it themselves by chance.

Reviewer: ```suggestion
Note that submitting a new "device support request" does not guarantee that someone else will develop a custom "quirk" for ZHA. The project relies on volunteering developers. However, without "device support requests", the developers may not be aware that your specific Zigbee device is not working correctly in ZHA. 
```

Reviewer: In some instances, I took the liberty of heavy editing. I think sometimes, less is more. At least for me, this level of detail can be overwhelming. For example, when near-synonyms (such as features and functionality) were used, I removed one because I find it easier to read. At the same time, I don't think much is lost when removing one or the other. Let me know what you think.
